app/src/test/java/ch/hikemate/app/model/elevation/ElevationRepositoryTest.kt

```diff
@@ -0,0 +1,253 @@
+package ch.hikemate.app.model.elevation
+
+import ch.hikemate.app.model.route.LatLong
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.junit.Assert.assertEquals
+import org.junit.Assert.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+@OptIn(ExperimentalCoroutinesApi::class)
+class ElevationRepositoryTest {
+  private lateinit var client: OkHttpClient
+  private lateinit var elevationServiceRepository: ElevationServiceRepository
+  private val longList =
+      listOf(515.0, 545.0, 117.0, 620.0, 480.0, 200.0, 750.0, 1020.0, 890.0, 1500.0, 340.0, 670.0)
+  private val smallList = listOf(515.0, 545.0, 117.0)
+
+  private val latLongList =
+      listOf(LatLong(10.0, 10.0), LatLong(20.0, 20.0), LatLong(41.161758, -8.583933))
+
+  private val longLatLongList =
+      listOf(
+          LatLong(10.0, 10.0),
+          LatLong(20.0, 20.0),
+          LatLong(41.161758, -8.583933),
+          LatLong(25.5, 15.5),
+          LatLong(35.0, 30.0),
+          LatLong(45.0, -5.0),
+          LatLong(50.0, 50.0),
+          LatLong(60.5, 60.5),
+          LatLong(40.0, -70.0),
+          LatLong(50.0, 90.0),
+          LatLong(55.0, 100.0),
+          LatLong(30.0, -120.0),
+      )
+
+  private val longResponse =
+      Response.Builder()
+          .code(200)
+          .message("OK")
+          .body(
+              """
+{
+   "elevations":
+   [
+      515.0,
+      545.0,
+      117.0,
+      620.0,
+      480.0,
+      200.0,
+      750.0,
+      1020.0,
+      890.0,
+      1500.0,
+      340.0,
+      670.0
+   ]
+}
+            """
+                  .trimIndent()
+                  .replace("\n", "")
+                  .toResponseBody("application/json; charset=utf-8".toMediaType()))
+          .protocol(Protocol.HTTP_1_1)
+          .header("Content-Type", "application/json")
+          .request(mockk())
+          .build()
+
+  private val simpleResponse =
```

Comment:
Nitpick: improve coherence in the naming : use long as prefix for all long dummy data list, and choose between simple and small for the shorter dummy data lists. Add a prefix to `latLongList`.

Reply:
Yep, would make sense 😂